library/Mockery/Generator/Parameter.php

```diff
@@ -56,7 +62,13 @@ public function getTypeHintAsString()
         }
 
         if (preg_match('/^Parameter #[0-9]+ \[ \<(required|optional)\> (?<typehint>\S+ )?.*\$' . $this->rfp->getName() . ' .*\]$/', $this->rfp->__toString(), $typehintMatch)) {
+
             if (!empty($typehintMatch['typehint'])) {
+
+                if ($this->isNamespacedClass($typehintMatch['typehint'])) {
```

Comment:
Ha, look few lines above in this class. There is already such code, that I guess needs to be extracted in separate method:
![typehinttextmethod](https://cloud.githubusercontent.com/assets/1277526/6199296/c8fe2814-b443-11e4-98ce-d09da4234999.png)